client/src/__tests__/callItEven.test.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event'
+import { navigate } from '@reach/router';
+import '@testing-library/jest-dom/extend-expect'
+import CallItEven from '../pages/CallItEven';
+import { postTransaction } from '../services/TransactionService';
+
+jest.mock('../services/TransactionService');
+postTransaction.mockResolvedValue('upload');
+
+jest.mock('@reach/router')
+navigate.mockResolvedValue('');
+
+const users =
+{
+  _id: "5f75e077dd559746a01dda55",
+  lead: "C",
+  leadEmail: "1234@test",
+  partner: "M",
+  partnerEmail: "123@test"
+};
+const summary = { balance: 5, totalOwed: 5, overallLender: "C" }
+const currency = '£';
+
+describe('Call it even', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  })
+  it('displays correct summary and users partner', async () => {
+    render(<CallItEven users={users} currency={currency} summary={summary} />);
+    expect(screen.getByText(/M\s*owes you\s*£5/)).toBeInTheDocument();
+  })
+  //mock form input - check (wipe that debt) naviagtes away from page
+  //wipe that debt also adjusts transaction
+  it('On "wipe that debt" click navigation is called', async () => {
+    render(<CallItEven
+      users={users}
+      currency={currency}
+      summary={summary}
+      setTransactions={val => {
+        console.log('transaction', val);
+      }}
+      isLoading={false}
+      setIsLoading={val => {
+        console.log('isloading', val);
+      }} />);
+    const a = await screen.getByText('Wipe that debt');
+    userEvent.click(a);
+    expect(navigate).toHaveBeenCalledTimes(1);
+  })
+  // check a new transaction is created on click?!
+  it('New transaction created on click', async () => {
+    // nothing to mock before hand
+    // apart from form entry
+    render(<CallItEven
+      users={users}
+      currency={currency}
+      summary={summary}
+      setTransactions={val => {
+        console.log('transaction', val);
+      }}
+      isLoading={false}
+      setIsLoading={val => {
+        console.log('isloading', val);
+      }} />);
+
+    const LeaveNote = screen.getByRole('LeaveNote');
```

Comment:
You could use .getByRole('textbox', {name: NAME})
You'd have to add an aria label, but I'd argue at least that's useful for accessibility purposes